### PR TITLE
Test config precedence over db relationship

### LINKED ISSUE #785: add a new test to see when either sourceFields or targetFields is NOT NULL AND Foreign Key is ALSO present
add a new test to see when either sourceFields or targetFields is NOT NULL AND Foreign Key is ALSO present - we give precedence to the source and target fields specified in the config to perform the join.

_Originally posted by @Aniruddh25 in https://github.com/Azure/data-api-builder/pull/707#discussion_r964283045_

### PR DESCRIPTION
**Pull request by abhishekkumams**
## Why make this change?

- Closes #785 
  - Test that source and target field specified in the config overrides the relationship fields defined in DB.

## What is this change?

- Adding a test consisting of two tables `players` and `clubs`, where the players table contain a coulmns like `current_club_id` and `new_club_id`. `current_club_id` is used as foriegn key for relationship with table `clubs`. 
- The test verifies the differnece how providing source fields as `new_club_id` and target field  as `id` in the config, overrrides the original relationship fields between `players` and `clubs`. 

## How was this tested?

- [x] Integration Tests

### CODE REVIEW

File: src/Service.Tests/SqlTests/GraphQLQueryTests/GraphQLQueryTestBase.cs [hunk 1]
```diff
@@ -1450,6 +1458,92 @@ public virtual async Task TestInvalidFilterParamQuery()
...
+        [TestMethod]
+        public async Task TestConfigTakesPrecedenceForRelationshipFieldsOverDB(
+            string[] sourceFields,
+            string[] targetFields,
+            int club_id,
+            string club_name,
+            DatabaseType dbType,
+            string testEnvironment)
+        {
+            RuntimeConfig configuration = ConfigurationTests.InitBasicRuntimeConfigWithNoEntity(dbType, testEnvironment);
```
Comment (Aniruddh25) [inline src/Service.Tests/SqlTests/GraphQLQueryTests/GraphQLQueryTestBase.cs:1475]: `InitBasicRuntimeConfigWithNoEntity` is a test helper method so better placement would be in `SqlTestHelper`

File: src/Service.Tests/SqlTests/GraphQLQueryTests/GraphQLQueryTestBase.cs [hunk 2]
```diff
@@ -1450,6 +1458,92 @@ public virtual async Task TestInvalidFilterParamQuery()
...
+            Entity clubEntity = new(
+                Source: JsonSerializer.SerializeToElement("clubs"),
+                Rest: true,
+                GraphQL: true,
+                Permissions: new PermissionSetting[] { ConfigurationTests.GetMinimalPermissionConfig(AuthorizationResolver.ROLE_ANONYMOUS) },
+                Relationships: null,
+                Mappings: null
+            );
+
+            configuration.Entities.Add("Club", clubEntity);
```
Comment (Aniruddh25) [inline src/Service.Tests/SqlTests/GraphQLQueryTests/GraphQLQueryTestBase.cs:1486]: Since we are adding these entities explicitly for the test, why do we need to modify the pipeline configs through config-generators/checked in config files?

**Review by Aniruddh25 — APPROVED** [review]
Thanks for adding this test! LGTM after resolving comments

File: src/Service.Tests/SqlTests/GraphQLQueryTests/GraphQLQueryTestBase.cs (hunk 2 above)
Comment (abhishekkumams) [reply src/Service.Tests/SqlTests/GraphQLQueryTests/GraphQLQueryTestBase.cs:1486]: it wasn't required. I did it just for consistency. but It makes sense to keep changes minimilistic.

**Review by ayush3797 — APPROVED** [review]
